src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapper.tsx

```diff
@@ -0,0 +1,158 @@
+import { GeneOptionLabel } from 'pages/resultsView/enrichments/EnrichmentsUtil';
+
+interface IGroupComparisonPathwayMapperProps {
+    genesOfInterest: AlterationEnrichmentRow[];
+    activeGroups: ComparisonGroup[] | undefined;
+    genomicData: AlterationEnrichmentRow[];
+    store: GroupComparisonStore;
+}
+
+@observer
+export default class GroupComparisonPathwayMapper extends React.Component<
+    IGroupComparisonPathwayMapperProps
+> {
+    clearMessage: (() => void) | undefined;
+
+    constructor(props: IGroupComparisonPathwayMapperProps) {
+        super(props);
+        makeObservable(this);
+
+        // @ts-ignore
+        import(/* webpackChunkName: "pathway-mapper" */ 'pathway-mapper').then(
+            (module: any) => {
+                this.PathwayMapperComponent = (module as any)
+                    .default as PathwayMapper;
+            }
+        );
+    }
+
+    @observable.ref PathwayMapperComponent:
+        | PathwayMapper
+        | undefined = undefined;
+    @observable isGeneSelectionPopupVisible: boolean = false;
+    @observable _genesOfInterest: string[] | undefined = undefined;
+
+    @computed get selectedGenes() {
+        return (
+            this._genesOfInterest ||
+            this.props.store.genesWithMaxFrequency.map(g => g.hugoGeneSymbol)
+        );
+    }
+
+    @computed get genesOfInterest(): AlterationEnrichmentRow[] {
+        return this.props.store.alterationEnrichmentRowData.filter(gene => {
+            return this.selectedGenes.includes(gene.hugoGeneSymbol);
+        });
+    }
+
+    public render() {
+        if (!this.PathwayMapperComponent) {
+            return null;
+        }
+
+        return (
+            <div>
+                <div className="alert alert-info">
+                    Ranking is based on top 10 genes having highest frequency by
+                    default. Genes of interest can be changed.
+                    <DefaultTooltip
+                        trigger={['click']}
+                        destroyTooltipOnHide={true}
+                        visible={this.isGeneSelectionPopupVisible}
+                        onVisibleChange={visible => {
+                            this.isGeneSelectionPopupVisible = !this
+                                .isGeneSelectionPopupVisible;
+                        }}
+                        overlay={
+                            <GenesSelection
+                                options={[]}
+                                selectedOption={{
+                                    label: GeneOptionLabel.USER_DEFINED_OPTION,
+                                    value: this.selectedGenes.join('\n'),
+                                }}
+                                onSelectedGenesChange={(
+                                    value,
+                                    genes,
+                                    label
+                                ) => {
+                                    this._genesOfInterest = genes.map(
```

Comment:
this should be put into a callback with @action decorator since it updates to pieces of state

Reply:
moved to a separate `onSelectedGenesChange` callback function and added `@action.bound` decorator